### PR TITLE
[SYCL] Add tests for new SYCL config processing

### PR DESCRIPTION
**Pull request by denis-kabanov**
Add nine tests for new SYCL config processing.

### CODE REVIEW

File: SYCL/Config/check_config_processing.cpp [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+  if (envir == "CHECK") {
+    try {
+      auto device = sycl::device{sycl::default_selector{}};
+    } catch (sycl::exception &e) {
+      std::string errorMessage = e.what();
+      if ((errorMessage == "SPACE found at the beginning/end of the line or "
+                           "before/after '='") ||
+          (errorMessage ==
+           "Variable name is more than 256 or less than one character") ||
```
Comment (dm-vodopyanov) [inline SYCL/Config/check_config_processing.cpp]: It's better to check for specific error message for each individual run.

File: SYCL/Config/check_config_processing.cpp [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+          (errorMessage == "The value contains more than 256 characters or "
```
Comment (dm-vodopyanov) [inline SYCL/Config/check_config_processing.cpp]: Need to skip comparison with the specific number during checking of error message (e.g., using regex)

File: SYCL/Config/check_config_processing.cpp [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  }
+  if (envir == "SMALL_NAME") {
+    std::cout << "=b" << std::endl;
+    return 0;
+  }
+  if (envir == "BIG_VALUE") {
+    // Max variable value contains 256 or 1024 characters
+    std::cout
+        << "a="
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
```
Comment (dm-vodopyanov) [inline SYCL/Config/check_config_processing.cpp]: Better to replace it here and above with a loop

**Review by vladimirlaz — CHANGES_REQUESTED** [review]
I suggest to abandon this PR because the test does not depend on BEs. It is better to have it as unittest in compiler repository.

**Pull request closed without merging**